### ISSUE TITLE
Add support for servers with non-ASCII response headers

### ISSUE THREAD

**Issue by yunfandev** [microsoft/reverse-proxy#440]
### Describe the bug

When I try to create a reverse proxy for the blog with the following response header:

```text
access-control-allow-credentials: true
access-control-allow-origin: *
cache-control: public, max-age=0
content-encoding: gzip
content-type: text/html; charset=utf-8
date: Fri, 25 Sep 2020 10:15:43 GMT
etag: W/"9127-u+4Jv+RfD2erbftmcMqr2DT5eig"
status: 304
strict-transport-security: max-age=315360000; includeSubDomains; preload
vary: Origin
x-powered-by: traefik(made with ❤️ with go)
```

It throws an expectation, `InvalidOperationException: Invalid non-ASCII or control character in header: 0x00E2` in line 698 😂😂😂

https://github.com/microsoft/reverse-proxy/blob/1b3a0a19b16826d3b75f994e7db4194c42379c78/src/ReverseProxy/Service/Proxy/HttpProxy.cs#L689-L699

### To Reproduce

```json
  "ReverseProxy": {
    "Routes": [
      {
        "RouteId": "blog",
        "ClusterId": "blog",
        "Match": {
          "Path": "{**catch-all}"
        }
      }
    ],
    "Clusters": {
      "blog": {
        "Destinations": {
          "blog/destination1": {
            "Address": "https://yunfan.dev/"
          }
        }
      }
    }
  }
```

### Further technical details

-  Microsoft.ReverseProxy: 1.0.0-preview.5.20467.3
- Runtime
  ```text
  .NET Core SDK (reflecting any global.json):
  Version:   3.1.402
  Commit:    9b5de826fd

  Runtime Environment:
  OS Name:     Windows
  OS Version:  10.0.19042
  OS Platform: Windows
  RID:         win10-x64
 ```

**Comment by yunfandev** [comment]
This seems to be a known issue. Are there any plans for improvement?

https://github.com/microsoft/reverse-proxy/issues/154#issuecomment-627572745

**Comment by Tratcher** [comment]
The HTTP RFCs fail to define a standard header encoding so any non-ascii characters will cause compatibility problems.

In this example that error is coming from [Kestrel](https://github.com/dotnet/aspnetcore/blob/e0467560e11a739897521326b543e9652a4edb77/src/Servers/Kestrel/Core/src/CoreStrings.resx#L226), which doesn't allow non-ascii response headers under any circumstances. I can't even find an open issue requesting support for it.

Recommendations:
1. Turn off or modify the x-powered-by header in the destination server if you control it.
1. Use a response header transform to remove/replace the x-powered-by header. See https://microsoft.github.io/reverse-proxy/articles/transforms.html#responseheader
1. File an issue against Kestrel in https://github.com/dotnet/aspnetcore. Note the next opportunity to address that will be 6.0.
1. Use HttpSys, it supports UTF-8 response header encoding.

**Comment by samsp-msft** [comment]
Oh dang, this is probably going to be a problem. If the destination we proxy to returns non ascii, then we need to be able to pass that on.

**Comment by Tratcher** [comment]
For reference, the ❤️ in the above example is utf-8 encoded `\xe2\x9d\xa4\xef\xb8\x8f`. HttpClient assumes latin1 and casts the bytes to characters like this: `0x00E2`. Kestrel scans the string and fails on the first non-ascii character. If Kestrel also allowed/assumed latin1 response header encoding then the UTF-8 bytes would flow through transparently to the client, but it would appear garbled if you tried to examine them inside the proxy.

@Pilchie FYI, I don't know that this specific example qualifies for an AspNetCore 3.1 or 5.0 patch, but I expect more like it in the future.

**Comment by karelz** [comment]
Triage: Chris couldn't find it in traefik source code - we think it is deployment option / modification.
Relies on ASP.NET - which is now in Future. Please upvote both this issue and ASP.NET linked issue https://github.com/dotnet/aspnetcore/issues/26334 if you hit this. if there is enough hits, we will consider service it.

**Comment by Tratcher** [comment]
This has been implemented in Kestrel and #1346 tracks adding it to the docs.
